Documentation/ProjectFile/prj/process_variables/process_variable/boundary_conditions/boundary_condition/NonuniformDirichlet/t_mesh.md

```diff
@@ -0,0 +1,8 @@
+Path to the surface mesh vtu file.
+
+The surface mesh must contain a nodal integer-valued field (unsigned 64bit)
+named \c OriginalSubsurfaceNodeIDs. That field establishes the mapping between
+the nodes of the surface mesh to some notes in the bulk mesh.
+\warning It is not checked if the surface mesh and the bulk mesh correspond to each
```

Comment:
Actually, I think we should add that check, because it's rather cheap (I guess). I know, I didn't provide the check in the NonuniformNeumann case. But someday, somebody will get weird results, because the one took the wrong boundary mesh...

Reply:
I think it would be nice, if you can give an option to write all found and actually used boundary conditions to a vtu-file for comparison. In this way, the user can compare the used BCs to the intended ones.

Reply:
That's a good idea, but maybe that will give little additional insight, or not the complete picture:
* Nothing is "found": The nonuniform BCs pretty much work like the direct BCs of OGS5 (as far as I understand that OGS-5 feature): The concrete bulk node ids are present in the boundary mesh. The difference is, that in OGS-6 they are nicely wrapped in a vtu file 😄.
* Imagine you have `coarse_bulk_mesh.vtu` and `coarse_surface_mesh.vtu`. The simulation will just run fine. Now you refine the mesh and get `fine_bulk_mesh.vtu`, but you forget to reconstruct the boundary mesh. Now, if you are unlucky, the fine mesh has first all the nodes of the coarse mesh, and then the refined ones. If you then output the "actually used" BC surface mesh, it will look perfectly fine, unless you investigate it in detail. But you would have gotten the same information from the mesh file (`coarse_surface_mesh.vtu`) that you supplied. Admittedly, this example might be a bit artificial, and in the majority of cases, the additional output might be helpful. But again: Here, nothing is "found", you have to "find" the right BC surface during preprocessing.

Maybe the feature you mentioned could be implemented as a preprocessing tool (`checkBoundaryMeshCorrespondsToBulkMesh`).

Reply:
Noted in https://github.com/ufz/ogs/issues/1915